crates/formality-check/src/mini_rust_check.rs

```diff
@@ -229,9 +228,7 @@ impl Check<'_> {
                 // Check if the value is well-formed.
                 let value_ty = self.check_value(typeck_env, switch_value).unwrap();
 
-                if !ty_is_int(value_ty) {
-                    bail!("The value used for switch must be an int.");
-                }
+                self.prove_goal(&typeck_env.env, Wcs::default(), value_ty.is_int())?;
```

Comment:
I personally don't like introducing a new judgement for this. I think if we want this check to be a proper judgement maybe add a *builtin trait* and prove `value_ty: BuiltinInteger` here :thinking: 

Even if it's a proper judegement:tm: but instead be just something else you can prove. Predicates feel *core* to the type system in a way in which "is this an integer" does not. That's just my personal vibe here though

What's the reasoning behind changing an assertion to an actual goal?

Reply:
could we instead just use a "non-predicate" judgement here. Same as we do for `trait_ref_is_knowable` or however it's called?